### ISSUE TITLE
Pluto TV: Playback hangs after playing channels for about 1 minute

### ISSUE THREAD

**Issue by ShaunTheSalvo** [matthuisman/slyguy.addons#791]
Hello,

Firstly wanted to say thank you for such a great collection of very useful Kodi addons!

I've been having some trouble playing back Pluto TV channels recently (about the last week or so). I watch a few music channels, in particular Now 80's in Pluto TV/UK. All other channels I've tried seem to have the same problem.

Occasionally, playback will not start when selecting a channel. Usually selecting the channel again after playback fails will get the channel playing. However, playback will hang roughly 1 minute into the stream, and will then hang and not resume playback. Stopping the stream and resuming after this occurs will result in the channel either failing to play, or (if it does play), repeating the hanging problem.

I have Kodi installed on both my Android TV box (H96 Max), and my PC (Intel laptop running Arch Linux with Kodi), and experiencing the same problem. it seems to be limited to the Pluto TV addon. Other addons such as IPTV AU and IPTV NZ seem to work with no problems.

Using "Alternative Streams" seems to reduce or eliminate the issue, testing just now, so that may be a workaround.

Thanks again for your hard work on these addons, they are awesome!

**Comment by matthuisman** [comment]
- probably due to server side ad injection and kodis inpustream adaptive addon still not working with x-discontinuity streams. pretty much out of my hands.
you could try disable inpustream adaptive for hls in addon settings. but i dont think ffmpeg handles them any better

**Comment by ShaunTheSalvo** [comment]
Wow - thank you for repling so quickly! :)

I've just now tried disabling inputstream adaptive for hls on my PC, and that seems to be working so far. I haven't tried it on the Android box yet, but will do so, probably tomorrow. I had a feeling it may be due to something on Pluto's side.

I found another possible workaround in using Samsung TV Plus instead of Pluto TV, which seems to also carry the channels I'm interested in, and so far is working fine, in fact channel lock-on when loading a channel seems to be faster. Thank you anyway for the response! Have a great day.
